.github/workflows/test.yml

```diff
@@ -15,4 +15,21 @@ jobs:
     - name: Checkout Code
       uses: actions/checkout@v3.3.0
     - name: Unit Tests
-      run: go test -v -race ./...
+      run: |
+        sudo go test ./... -coverprofile coverage.out -covermode count
+        sudo go tool cover -func coverage.out
+    - name: Quality Gate - Test coverage shall be above threshold
+      env:
+        TESTCOVERAGE_THRESHOLD: 75
+      run: |
+        echo "Quality Gate: checking test coverage is above threshold ..."
+        echo "Threshold             : $TESTCOVERAGE_THRESHOLD %"
+        totalCoverage=`go tool cover -func=coverage.out | grep total | grep -Eo '[0-9]+\.[0-9]+'`
+        echo "Current test coverage : $totalCoverage %"
+        if (( $(echo "$totalCoverage $TESTCOVERAGE_THRESHOLD" | awk '{print ($1 > $2)}') )); then
+            echo "OK"
+        else
+            echo "Current test coverage is below threshold. Please add more unit tests or adjust threshold to a lower value."
+            echo "Failed"
+            exit 1
+        fi
```

Comment:
Newline is missing at the end of the file.

Reply:
Also, I think `sudo` is not required. A reference of such tests from etcd [here][1].

[1]: https://github.com/etcd-io/etcd/blob/main/.github/workflows/tests.yaml